pennylane/tape/measure.py

```diff
@@ -315,6 +315,14 @@ def circuit():
 def state():
     r"""Quantum state in the computational basis.
 
+    .. note::
+
+        The quantum state can only be returned in tape mode:
```

Comment:
I'm not quite sure I follow your question @trbromley!

Both the following work for me:

```python
import pennylane as qml

dev = qml.device("default.qubit", wires=2)

@qml.tape.qnode(dev)
def circuit():
    qml.Hadamard(wires=1)
    return qml.tape.state()

print(circuit())
```

and 

```python
import pennylane as qml

qml.enable_tape()
dev = qml.device("default.qubit", wires=2)

@qml.qnode(dev)
def circuit():
    qml.Hadamard(wires=1)
    return qml.state()

print(circuit())
```

The only reason I mentioned only the `enable_tape` approach in the docstring was I thought it might be too confusing to discuss both approaches (the first one I consider to be a bit more 'advanced')